File: backend/app/schemas/highlight.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel
from .highlight_rect import HighlightRectRead
from .comment import CommentRead

class HighlightBase(BaseModel):
    document_file_id: int
    created_by: str  # 'user' | 'ai'
    memo: Optional[str] = None
    text: Optional[str] = None

class HighlightCreate(HighlightBase):
    # LLM生成ハイライト用メタデータ
    suggestion_reason: Optional[str] = None

class HighlightUpdate(BaseModel):
    memo: Optional[str] = None
    text: Optional[str] = None

class HighlightRead(HighlightBase):
    id: int
    comment_id: Optional[int] = None
    document_file_id: int
    created_by: str
    memo: str
    text: Optional[str] = None
    created_at: datetime
    rects: List[HighlightRectRead]

    class Config:
        from_attributes = True

class HighlightDelete(BaseModel):
    """ハイライト削除時のレスポンス"""
    message: str
    deleted_highlight_id: int
    deleted_comments_count: int
    deleted_rects_count: int

class HighlightWithComments(BaseModel):
    highlight: HighlightRead
    comments: List[CommentRead]
    
    class Config:
        from_attributes = True

File: frontend/types/Responses/Highlight.ts
import { HighlightRect, HighlightRectEntity } from './HighlightRect';
import { CommentResponse } from './Comment';

export interface HighlightEntity {
  id: number;
  project_file_id: number;
  created_by: string;
  memo: string;
  text: string | null;
  element_type: string;
  created_at: string;
  rects: HighlightRectEntity[];
}

export interface CreateHighlightResponse {
  id: number;
  comment_id: number;
  document_file_id: number;
  text: string;
  memo: string;
  created_by: string;
  created_at: string;
}

export interface HighlightDeleteResponse {
  message: string;
}

export interface HighlightResponse {
  id: number;
  document_file_id: number;
  text: string;
  memo: string;
  created_by: string;
  created_at: string;
  rects: HighlightRect[];
}

export interface HighlightWithCommentsResponse {
  highlight: HighlightResponse;
  comments: CommentResponse[];
}